### DISCUSSION TITLE
Which versions of Python should we support?

### DISCUSSION THREAD

**Discussion by proc-2** [techno-solutionist-collective/dreadlocks#27] [Polls]
Currently we do not support Python 3.6, Python 3.7, nor Python 3.8: newer versions of `poetry` are incompatible with `3.6` (`--only` option is missing in older versions for instance), `pip-licenses` requires `>= 3.8`, and Python 3.8 does not support simplified type hints available since Python 3.9 (such as applying generics to `dict` or `list` directly). Both 3.6 and 3.7 are past their "end-of-life", while 3.8 will enter "end-of-life" somewhere after mid-2024 (https://devguide.python.org/versions/). Nevertheless, I wonder if there is interest to widen the support of this library to older versions of Python.

**Comment by rikardn** [comment]
I think only officially supported versions of Python should be supported. That means Python 3.8 to 3.12 (Note that numpy (NEP 29) has dropped support for 3.9, but it is still an officially supported version). If support soon ends for 3.8 and 3.9 has introduced some nice features it might not be worth the hassle of supporting it.
